odoo/addons/base/models/ir_property.py

```diff
@@ -310,16 +310,20 @@ def clean(value):
         ])
 
         # modify existing properties
-        for prop in props:
-            id = refs.pop(prop.res_id)
+        ref_list = list(refs.keys())
+        for ref in ref_list:
+            prop = props.filtered(lambda p: p.res_id == ref)
```

Comment:
Turning an O(n) algorithm into an O(n²) algorithm for detecting an error that mostly never happens is not acceptable, come on.  Is this issue blocking?  I mean, reproducing the issue seems to require a user to voluntarily fuck up its database.  The user may remove the `ir.property` records on their own, may they not?

Reply:
Yeah, you are right.